pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -396,34 +397,7 @@ func ApplyResultsToWorkspaceBindings(trResults map[string][]v1.TaskRunResult, pr
 		}
 	}
 
-	for i, binding := range pr.Spec.Workspaces {
-		if pr.Spec.Workspaces[i].PersistentVolumeClaim != nil {
-			pr.Spec.Workspaces[i].PersistentVolumeClaim.ClaimName = substitution.ApplyReplacements(binding.PersistentVolumeClaim.ClaimName, stringReplacements)
-		}
-		pr.Spec.Workspaces[i].SubPath = substitution.ApplyReplacements(binding.SubPath, stringReplacements)
-		if pr.Spec.Workspaces[i].ConfigMap != nil {
-			pr.Spec.Workspaces[i].ConfigMap.Name = substitution.ApplyReplacements(binding.ConfigMap.Name, stringReplacements)
-		}
-		if pr.Spec.Workspaces[i].CSI != nil {
-			pr.Spec.Workspaces[i].CSI.Driver = substitution.ApplyReplacements(binding.CSI.Driver, stringReplacements)
-			if pr.Spec.Workspaces[i].CSI.NodePublishSecretRef != nil {
-				pr.Spec.Workspaces[i].CSI.NodePublishSecretRef.Name = substitution.ApplyReplacements(binding.CSI.NodePublishSecretRef.Name, stringReplacements)
-			}
-		}
-		if pr.Spec.Workspaces[i].Secret != nil {
-			pr.Spec.Workspaces[i].Secret.SecretName = substitution.ApplyReplacements(binding.Secret.SecretName, stringReplacements)
-		}
-		if pr.Spec.Workspaces[i].Projected != nil {
-			for j, source := range binding.Projected.Sources {
-				if pr.Spec.Workspaces[i].Projected.Sources[j].ConfigMap != nil {
-					pr.Spec.Workspaces[i].Projected.Sources[j].ConfigMap.Name = substitution.ApplyReplacements(source.ConfigMap.Name, stringReplacements)
-				}
-				if pr.Spec.Workspaces[i].Projected.Sources[j].Secret != nil {
-					pr.Spec.Workspaces[i].Projected.Sources[j].Secret.Name = substitution.ApplyReplacements(source.Secret.Name, stringReplacements)
-				}
-			}
-		}
-	}
+	pr.Spec.Workspaces = workspace.ReplaceWorkspaceBindingsVars(pr.Spec.Workspaces, stringReplacements)
```

Comment:
Refactor this section of the code so that it can be reused in taskrun and pipelinerun.